merklize/merklize.go

```diff
@@ -156,6 +160,44 @@ func NewFieldPathFromContext(ctxBytes []byte, ctxType, fieldPath string) (Path,
 	return resPath, nil
 }
 
+// TypeFromContext returns type of field from context by path.
+func TypeFromContext(ctxBytes []byte, path string) (string, error) {
+	var ctxObj map[string]interface{}
+	err := json.Unmarshal(ctxBytes, &ctxObj)
+	if err != nil {
+		return "", err
+	}
+
+	ldCtx, err := ld.NewContext(nil, nil).Parse(ctxObj["@context"])
+	if err != nil {
+		return "", err
+	}
+
+	parts := strings.Split(path, ".")
+
+	for _, term := range parts {
```

Comment:
can't you reuse pathFromContext function

Reply:
These functions look the same but use different logic.